backend/clap/proto/data_store.proto

```diff
@@ -0,0 +1,106 @@
+// Proto file from the shared folder
+syntax = "proto3";
+
+package data_store;
+
+option java_multiple_files = true;
+option java_outer_classname = "DataStoreProto";
+option go_package = "github.com/Veermove/zpi/data-store";
+
+enum ProjectState {
+    DRAFT = 0;
+    SUBMITTED = 1;
+    ACCEPTED = 2;
+    REJECTED = 3;;
+}
+
+enum RatingType {
+    INDIVIDUAL = 0;
+    INITIAL = 1;
+    FINAL = 2;
+}
+
+message SubmissionRequest {
+    int32 contest_id = 1;
+}
+
+message DetailsSubmissionRequest {
+    int32 submission_id = 1;
+}
+
+message RatingsSubmissionRequest {
+    int32 submission_id = 1;
+    int32 constest_id   = 2;
+}
+
+message SubmissionsResponse {
+    repeated Submission submissions = 1;
+}
+
+message Submission {
+    int32  submission_id = 1;
+    int32  year          = 2;
+    string name          = 3;
+    int32  duration_days = 4;
+
+    repeated Assessor assessors = 5;
+    repeated Rating   ratings   = 6;
+}
+
+message Assessor {
+    string first_name  = 1;
+    string last_name   = 2;
+    int32  assessor_id = 3;
+}
+
+message Rating {
+    int32      rating_id   = 1;
+    bool       is_draft    = 2;
+    int32      assessor_id = 3;
+    RatingType type        = 4;
+}
+
+message DetailsSubmissionResponse {
+    int32        team_size   = 1;
+    string       finish_date = 2;
+    ProjectState status      = 3;
+    string       budget      = 4;
+    string       description = 5;
+}
+
+message Criterion {
+    int32 criterion_id = 1;
+    string name        = 2;
+    string description = 3;
+    string area        = 4;
+    string criteria    = 5;
+    string subcriteria = 6;
```

Comment:
This is an optional field, can we mirror that fact in the proto file?

Reply:
Since protobuf [release 3.15](https://github.com/protocolbuffers/protobuf/releases/tag/v3.15.0), proto3 supports using the `optional` keyword, however i'm still using 3.14, where this feature is unavailable, and i'd rather not upgrade right now.

Either way it is not necessary, nor mandatory to create message with EVERY field from it's definition. 
In Go this results in default values for missing fields (so `0`,`""` or `nil` if its pointer to another struct).
Not sure about java, i would assume that missing fields are just `null`.

Reply:
okay, all the fields are optional by default afaik
I just wanted to have a better compliance with the design we have